scrapy/spider.py

```diff
@@ -31,11 +31,19 @@ def __init__(self, name=None, **kwargs):
         if not hasattr(self, 'start_urls'):
             self.start_urls = []
 
-    def log(self, message, level=log.DEBUG, **kw):
-        """Log the given messages at the given log level. Always use this
-        method to send log messages from your spider
+    @property
+    def logger(self):
+        logger = logging.getLogger(self.name)
```

Comment:
I like `scrapy.core.scraper` or `scrapy.contrib.logstats` names in @dangra's gist. It helps with debugging to see where the log (error) message came from.

But doesn't it mean TopLevelFormatter is not working?